subprojects/common/http-client/src/main/kotlin/com/avito/http/internal/StatsHttpEventListener.kt

```diff
@@ -0,0 +1,59 @@
+package com.avito.http.internal
+
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsDSender
+import com.avito.android.stats.TimeMetric
+import com.avito.time.TimeProvider
+import okhttp3.Call
+import okhttp3.EventListener
+import okhttp3.Request
+import okhttp3.Response
+import java.io.IOException
+import java.net.SocketTimeoutException
+
+internal class StatsHttpEventListener(
+    private val statsDSender: StatsDSender,
+    private val timeProvider: TimeProvider
+) : EventListener() {
+
+    var callStarted = 0L
```

Comment:
Shouldn't we use Factory to make it thread-safe (unique for each Call)?

https://square.github.io/okhttp/events/#eventlistenerfactory

Reply:
i am using factory